### ISSUE TITLE
Remove redundant parentheses isn't reported for single-parameterized lambda

### ISSUE THREAD

**Issue by DoctorKrolic** [dotnet/roslyn#61882]
**Version Used**:
https://github.com/dotnet/roslyn/commit/6d15eebbe642f0c1c3c903be9326261b996f0e28

**Steps to Reproduce**:
```cs
M((a) => { });

void M(Action<int> a)
{

}
```

**Expected Behavior**:
`IDE0047` "Remove redundant parentheses" is reported for `(a)`

**Actual Behavior**:
No diagnostic is reported

**Comment by CyrusNajmabadi** [comment]
We considered this, but rejected it in the past.  The general feeling was that this was really about removing just parenthesized-expressions that were not needed.  IN this case, this isn't a parenthesized expression.  This is just a stylistic choice on whether you want to write simple lambdas or parenthesized ones.  We would want a new code-style for this, and tehre just didn't seem to be any community interest in this one-off feature, so we decided not to do it.

If you want though, you could write your own analyzer/fixer combo for your own repos.
